### PR TITLE
Added new users command to cli

### PR DESCRIPTION
**Pull request by aashir21**
Signed-off-by: Aashir Siddiqui <aashir_sidiki@hotmail.com>

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: pkg/users/users.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+func validateLoginIdFlag(loginId string) error {
+
+	var err error
+
+	if loginId == "" || strings.TrimSpace(loginId) == "" {
+		err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_MISSING_USER_LOGIN_ID_FLAG)
+	}
+
+	if err == nil {
```
Comment (techcobweb) [inline pkg/users/users.go:23]: Can we trim the string and use the trimmed content ? currently we trim it to check for blank, but throw it away.

Comment (techcobweb) [reply pkg/users/users.go:23]: Probably have to return the modified loginId so the caller can use it.

File: pkg/users/usersGet.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	var err error
+	var outputText []byte
+
+	err = validateLoginIdFlag(loginId)
+	if err == nil {
+		outputText, err = getUserDataFromRestApi(loginId, apiClient)
+
+		if err == nil {
+			var userData []UserData
+			err = json.Unmarshal(outputText, &userData)
```
Comment (techcobweb) [inline pkg/users/usersGet.go:32]: This can be avoided. See below.

File: pkg/users/usersGet.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+	if err == nil {
+
+		apiCall := apiClient.UsersAPIApi.GetUserByLoginId(context).LoginId(loginId).ClientApiVersion(restApiVersion)
+		userProperties, _, err = apiCall.Execute()
+
+		if err == nil {
+			userPropertiesJSON, err := json.Marshal(userProperties)
+			if err == nil {
+				return userPropertiesJSON, err
```
Comment (techcobweb) [inline pkg/users/usersGet.go:67]: I don't like multiple return statements in a function.
Why do we pass back JSON bytes ? We have the user properties, can't we return those ? Then we wouldn't need to turn it back from JSON to a structure again.

**Review by techcobweb — CHANGES_REQUESTED** [review]
Some changes requested. But looks good.

**Comment by galasa-team** [comment]
Build successful

File: pkg/users/users.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+		if loginId != "me" {
+			err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_LOGIN_ID_NOT_SUPPORTED, loginId)
+		}
+	}
+
+	loginId = strings.TrimSpace(loginId)
```
Comment (techcobweb) [inline pkg/users/users.go:30]: Duplicate code of Trimspace with the above. Some moving around of the if logic will mean you only do that once.

File: pkg/users/usersGet.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/galasa-dev/cli/pkg/spi"
+)
+
+func GetUsers(loginId string, apiClient *galasaapi.APIClient, console spi.Console) error {
+	var err error
+	var outputText []galasaapi.UserData
+
+	loginId, err = validateLoginIdFlag(loginId)
+	if err == nil {
+		outputText, err = getUserDataFromRestApi(loginId, apiClient)
```
Comment (techcobweb) [inline pkg/users/usersGet.go:23]: outputText is oddly named. It's not text which is returned. It's a user

File: pkg/users/usersGet.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+			extractedUserObject := &outputText[0]
+			console.WriteString("id: " + extractedUserObject.GetLoginId())
+		}
+
+	}
+
+	return err
+}
+
+func getUserDataFromRestApi(
```
Comment (techcobweb) [inline pkg/users/usersGet.go:35]: As this method is called getUserData... I would expect it to return a single user data record, not an array.

**Review by techcobweb — APPROVED** [review]
Approved assuming the minor comments are sorted out.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-271-8rlrt for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**